frontend/class-opengraph.php

```diff
@@ -505,11 +512,15 @@ public function description( $echo = true ) {
 			}
 		}
 
-		if ( is_singular() ) {
-			$ogdesc = WPSEO_Meta::get_value( 'opengraph-description' );
+		$is_posts_page = is_home() && ! is_front_page();
+
+		if ( is_singular() || $is_posts_page ) {
+			$post_id = ( $is_posts_page ) ? get_option( 'page_for_posts' ) : get_the_ID();
+			$post    = get_post( $post_id );
```

Comment:
@Rarst maybe you could do something about the duplication here?

Perhaps `$is_posts_page`, `$post_id` and `$post` could be properties that are already set in the constructor?